_docs/step-development-guideline.md

```diff
@@ -1,5 +1,45 @@
 # Step Development Guideline
 
+## Step dependencies
+
+In the context of Steps, dependency means an executable available in the PATH environment variable, which is executed by the step.
+
+`deps` should not include an executable needed to run the step.
```

Comment:
This only applies for `go` steps (and `bash` steps) but this sentence sounds like it would apply for any step, e.g. if you write a Ruby or Node step. In those cases you still have to install Ruby or Node, so those have to be in `deps`.

Reply:
I'd phrase it like this:

`deps` should not include toolkit dependencies, as the Bitrise CLI will take care of installing those automatically.